pkg/apis/storage/validation/validation.go

```diff
@@ -442,6 +443,29 @@ func validatePodInfoOnMount(podInfoOnMount *bool, fldPath *field.Path) field.Err
 	return allErrs
 }
 
+// validateSupportsFSGroup tests if SupportsFSGroup contains an appropriate value.
+func validateSupportsFSGroup(supportsFSGroup *storage.SupportsFSGroup, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if supportsFSGroup == nil {
+		// This is not a required field, so if nothing is provided simply return
+		return allErrs
+	}
+	switch *supportsFSGroup {
```

Comment:
We generally use sets to validate the strings: https://github.com/kubernetes/kubernetes/blob/f8493d70b26dd89ce21e4796e626b0706d557b11/pkg/apis/storage/validation/validation.go#L115

I know the volumelifecyclemode below doesn't do this, and perhaps we should change that (in a separate pr) to be consistent.

Reply:
I submitted https://github.com/kubernetes/kubernetes/pull/92887 to update the VolumeLifecycleModes validation section.